xmr-btc/tests/tor.rs

```diff
@@ -0,0 +1,69 @@
+#[cfg(feature = "tor")]
+mod tor_test {
+
+    use anyhow::Result;
+    use hyper::service::{make_service_fn, service_fn};
+    use reqwest::StatusCode;
+    use spectral::prelude::*;
+    use std::convert::Infallible;
+    use tokio::sync::oneshot::Receiver;
+    use torut::onion::TorSecretKeyV3;
+    use xmr_btc::tor::{AuthenticatedConnection, TOR_PROXY_ADDR};
+
+    async fn hello_world(
+        _req: hyper::Request<hyper::Body>,
+    ) -> Result<hyper::Response<hyper::Body>, Infallible> {
+        Ok(hyper::Response::new("Hello World".into()))
+    }
+
+    fn start_test_service(port: u16, rx: Receiver<()>) {
+        let make_svc =
+            make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(hello_world)) });
+        let addr = ([127, 0, 0, 1], port).into();
+        let server = hyper::Server::bind(&addr).serve(make_svc);
+        let graceful = server.with_graceful_shutdown(async {
+            rx.await.ok();
+        });
+        tokio::spawn(async {
+            // server.await.unwrap();
+            if let Err(e) = graceful.await {
+                eprintln!("server error: {}", e);
+            }
+        });
+    }
+
+    #[tokio::test]
+    async fn test_tor_control_port() -> Result<()> {
+        // Setup test HTTP Server
+        let (tx, rx) = tokio::sync::oneshot::channel::<()>();
+        let port = 8080;
+        start_test_service(port, rx);
+
+        // Connect to local Tor service
+        let mut authenticated_connection = AuthenticatedConnection::new().await?;
```

Comment:
:thinking: I guess this means that we assume that Tor is running before this test is run. We should skip this in CI and document it if that's the case.

Reply:
I'm starting Tor from the test now. For that it assumes that Tor is in the path. 
Not sure if this is the nicest way to go. Maybe mentioning it in the readme is enough? 
It's behind a feature flag anyway.

Reply:
I'm starting Tor from the test now. For that it assumes that Tor is in the path. 
Not sure if this is the nicest way to go. Maybe mentioning it in the readme is enough? 
It's behind a feature flag anyway.